### ISSUE TITLE
Support third party ESM loader

### ISSUE THREAD

**Issue by iFwu** [fgnass/node-dev#242]
When using the `experimental-loader` option, auto use internal ESM loader for the file.
```
node-dev --experimental-loader ts-node/esm/transpile-only ./index.ts
```

**Comment by iFwu** [comment]
Maybe need Node.js chained resolve hooks to be finished. The Node.js module loader is still highly experimental. https://github.com/nodejs/node/issues/36396

**Comment by bjornstar** [comment]
Yeah, we can only use a single module loader at the moment. I think it was a bit premature to attempt to use it.
